client/validators.py

```diff
@@ -1,9 +1,18 @@
 import re
 from django.core.exceptions import ValidationError
+from django.core.validators import RegexValidator
 
 ID_NUMBER_REGEX = r'^\d{8,9}$'
 
 
+# Checks if the phone number starts with +972/0, and then if the second part is one of:
+# 1. a land line geographic based number - 2/3/4/8/9 and afterwards 7 digits.
+# 2. a new number country prefix - 7 and afterwards 8 digits.
+# 3. a mobile number - 5 and afterwards 8 digits.
+PHONE_NUMBER_REGEX = re.compile(r"^\+?(972|0)(-)?0?(([23489]\d{7})|[5|7]\d{8})$")
```

Comment:
Could be written as: `^(\+972-?|0)(([23489]-?\d{7})|[5|7]\d-?\d{4}-?\d{3})$`

I changed the beginning of the regex as it seemed too complicated to check for +972-? or 0, and added hyphen support.

Reply:
Hyphens should also be optional.

Reply:
Added hyphens support as optional. Also added the option to have whitespace after +972.